[inductor] fix triton bucketize mask propagation

v0i0 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #158462
* __->__ #159961

See https://hud.pytorch.org/pytorch/pytorch/commit/6b414f56a4a133a428af618d8ed1553849341497

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @coconutruben

v0i0 committed:
[inductor] fix triton bucketize mask propagation

See https://hud.pytorch.org/pytorch/pytorch/commit/6b414f56a4a133a428af618d8ed1553849341497

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:
ghstack-source-id: 23c0501843caf936de9e534744b7b47f9125750a
Pull-Request: https://github.com/pytorch/pytorch/pull/159930

[ghstack-poisoned]

v0i0 committed:
Update on "[inductor] fix triton bucketize mask propagation"


See https://hud.pytorch.org/pytorch/pytorch/commit/6b414f56a4a133a428af618d8ed1553849341497

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:
Pull-Request: https://github.com/pytorch/pytorch/pull/159930

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

davidberard98 commented on test/inductor/test_torchinductor.py:
```diff
@@ -13654,6 +13654,28 @@ def forward(self, x):
         self.common(Model(), inputs)
 
+    @unittest.skipIf(
+        config.cpp_wrapper,
+        "cpp_wrapper samples will lead to invalid indexing",
```
what happens when cpp_wrapper is used - does it raise a device-side assert? (and is it during compilation or during runtime?) and do you know if that is a regression or if that is a previously-existing bug?

v0i0 replied:
cpp wrapper generates examples and runs kernels using those examples. these examples have the rule - random tensor for real data, zero data for integral data (to capture the kernel cubin i think?). so if the model makes assumptions (like repeats.sum() == output_size), cpp_wrapper fails. this is why the original MR disabled the decomp for cpp_wrapper, too.

v0i0 committed:
Update on "[inductor] fix triton bucketize mask propagation"


See https://hud.pytorch.org/pytorch/pytorch/commit/6b414f56a4a133a428af618d8ed1553849341497

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:
Pull-Request: https://github.com/pytorch/pytorch/pull/159930

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

v0i0 committed:
Update on "[inductor] fix triton bucketize mask propagation"


See https://hud.pytorch.org/pytorch/pytorch/commit/6b414f56a4a133a428af618d8ed1553849341497

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:
Pull-Request: https://github.com/pytorch/pytorch/pull/159930

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov coconutruben

[ghstack-poisoned]

v0i0 committed:
Update on "[inductor] fix triton bucketize mask propagation"


See https://hud.pytorch.org/pytorch/pytorch/commit/6b414f56a4a133a428af618d8ed1553849341497

Summary:

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:
Pull-Request: https://github.com/pytorch/pytorch/pull/159930

[ghstack-poisoned]

eellison commented on torch/_inductor/codegen/triton.py:
```diff
@@ -2720,6 +2720,8 @@ def bucketize(
             dtype=indexing_dtype,  # type: ignore[attr-defined]
         )
 
+        result.mask_vars = values.mask_vars  # type: ignore[attr-defined]
```
Would we need to add the boundaries mask as well ?

v0i0 replied:
added all CSEVariables

v0i0 committed:
Update

[ghstack-poisoned]

v0i0 commented:
@pytorchbot merge

v0i0 committed:
Update

[ghstack-poisoned]

v0i0 commented:
@pytorchbot merge
